php/libraries/Permission.class.inc

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * This file contains the Permission class.
+ *
+ * PHP Version 5-7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Rida Abou-Haidar <rida.loris@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * The Loris Permission class
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Rida Abou-Haidar <rida.loris@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Permission
+{
+    /**
+     * Stores Database being used
+     *
+     * @var    $DB Database
+     * @access private
+     */
+    var $DB;
+
+    /**
+     * Permission constructor.
+     *
+     * @param Database $Database database
+     */
+    function __construct($Database)
+    {
+        $this->DB = $Database;
+    }
+
+
+    /**
+     * Gets the list of permissions in the database.
+     *
+     * @return array Associative array in the form $permissionID=>$permissionName
+     */
+    function getPermissions()
+    {
+        $permissions = $this->DB->pselectColWithIndexKey(
+            "SELECT permID, code
+         FROM permissions",
+            array(),
+            "permID"
+        );
+
+        return $permissions;
+    }
+
+    /**
+     * Gets the list of permission labels in the database.
+     *
+     * @return array Associative array in the form $permissionID=>$permissionLabel
+     */
+    function getPermissionLabels()
+    {
+        $permissions = $this->DB->pselectColWithIndexKey(
+            "SELECT permID, description
+         FROM permissions",
+            array(),
+            "permID"
+        );
+
+        return $permissions;
+    }
+
+    /**
+     * Checks if the string is a permission within the database
+     *
+     * @param string $permissionName the permission to be checked
+     *
+     * @return boolean
+     */
+    function isPermission($permissionName)
```

Comment:
Shouldn't a permission class instance always be a permission?

Reply:
@driusan no, this is not a model, just a DAO i.e. an instance of a permission class does not represent a permission in this case, just a tool which knows how to deal with permissions

Reply:
then it's misleading to call it "Permission" in the root namespace... like I said in the summary, once there's a better class doc I can do a better review and/or suggest another name.